pkg/ebpftracer/c/tracee.bpf.c

```diff
@@ -6466,6 +6468,50 @@ statfunc bool net_l7_is_socks5(struct __sk_buff *skb, u32 l7_off)
 out:
     return false;
 }
+
+// see https://datatracker.ietf.org/doc/html/rfc4253 for the definition of the ssh protocol
+statfunc bool net_l7_is_ssh(struct __sk_buff *skb, u32 l7_off)
+{
+    char buf[ssh_min_len];
+    __builtin_memset(&buf, 0, sizeof(buf));
+
+    if (skb->len < l7_off) {
+        return false;
+    }
+
+    u32 payload_len = skb->len - l7_off;
+    u32 read_len = payload_len;
+    // inline bounds check to force compiler to use the register of size
+    asm volatile("if %[size] < %[max_size] goto +1;\n"
+                 "%[size] = %[max_size];\n"
+                 :
+                 : [size] "r"(read_len), [max_size] "i"(ssh_min_len));
+
+    // make the verifier happy to ensure that we read more than a single byte
+    asm goto("if %[size] < %[min_ssh] goto %l[out]"
+             :
+             : [size] "r"(read_len), [min_ssh] "i"(ssh_min_len)::out);
+
+    if (read_len < ssh_min_len) {
+        return false;
+    }
+
+    // load first ssh_min_len bytes from layer 7 in packet.
+    if (bpf_skb_load_bytes(skb, l7_off, buf, read_len) < 0) {
+        return false; // failed loading data into http_min_str - return.
+    }
+
+    // the rfc mentions that a server could also send other data before the ssh version, meaning
```

Comment:
very good to note this here. Maybe we should document this somewhere else as well? 🤔

Reply:
Mhm where would you expect such a piece of info?

Reply:
Afaik, we currently have no such place. But I guess close to our rules then. I think, at some point we should provide documentation for our default ruleset